compiler/src/iree/compiler/Codegen/LLVMGPU/KernelConfig.cpp

```diff
@@ -179,6 +179,14 @@ static bool supportsTensorCore(func::FuncOp entryPoint, linalg::LinalgOp op,
 static LogicalResult setContractConfig(func::FuncOp entryPoint,
                                        linalg::LinalgOp op,
                                        const TargetInfo &targetInfo) {
+  if (!linalg::isaContractionOpInterface(op) || op.getNumParallelLoops() < 2)
+    return failure();
+  // Don't consider operations that don't have a broadcast, those should go
+  // through reductions.
+  if (llvm::any_of(op.getIndexingMapsArray(),
+                   [](AffineMap m) { return m.isPermutation(); }))
```

Comment:
While I'm fine with this, it seems somewhat confusing that checking for permutations inherently equals "broadcast". It seems more likely you want to check index maps with a non-unique mappings? (i.e. 1, 2, 1). 

is there a direct way to check for this?